From 8a4090f6ab698c205d9df6a84b8273de4632878a Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 24 Feb 2025 16:58:59 +0100
Subject: [PATCH] arm64: dts: st: change power domain for mdf-dai on stm32mp21
 dk board

The MDF peripheral supports LP-Stop1/2 low power mode.
(See "Functionalities depending on system operating mode" table in ref
manual). However, the mdf driver does not support the autonomous mode.
Change power domain to "d1_pd" for mdf-dai sub-nodes, on stm32mp21 dk
board device tree.

Fixes: aeb48a517671 ("arm64: dts: st: change power domain for mdf on stm32mp211.dtsi")
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id54310555c927fc96df5a6ad8447665fc0a5fadd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/439591
Reviewed-by: Olivier MOYSAN <olivier.moysan@foss.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp215f-dk.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
index 793f67b8aa4a..f920bd050a23 100644
--- a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
@@ -456,7 +456,7 @@ asoc_pdm0: mdf-dai {
 			compatible = "st,stm32mp25-mdf-dai";
 			#sound-dai-cells = <0>;
 			io-channels = <&filter0 0>;
-			power-domains = <&ret_pd>;
+			power-domains = <&d1_pd>;
 			status = "okay";
 			mdf1_port0: port {
 				mdf_endpoint0: endpoint {
@@ -483,7 +483,7 @@ asoc_pdm1: mdf-dai {
 			compatible = "st,stm32mp25-mdf-dai";
 			#sound-dai-cells = <0>;
 			io-channels = <&filter1 0>;
-			power-domains = <&ret_pd>;
+			power-domains = <&d1_pd>;
 			status = "okay";
 
 			mdf1_port1: port {
-- 
2.50.1

